server.py

```diff
@@ -119,25 +120,31 @@ def data_loader(path, context):
         for content_fname in glob.glob(os.path.join(path, "*.*")):
             if (content_fname.endswith("yaml") or content_fname.endswith("json")):
                 continue
-            content_path = os.path.join(content_fname)
-            # print("Content file: {}, properties: {}".format(cf, str(context)))
-            entries.append(Image(content_path, context))
+            # csv files contain a list of paths to image files to be loaded
+            elif content_fname.endswith("csv"):
+                with open(os.path.join(content_fname), 'r') as list_file:
+                    for f in list_file:
+                        f = f.replace("\n", "")
+                        entries.append(Image(f, context))
+            else:
+                content_path = os.path.join(content_fname)
+                # print("Content file: {}, properties: {}".format(cf, str(context)))
+                entries.append(Image(content_path, context))
 
         # finally, we iterate down, copying the context.
         for root, dirs, files in os.walk(path):
             for d in dirs:
```

Comment:
Python's `os.walk` doesn't have a well defined ordering, it's the ordering as on the file system. So we should probably do a `for d in sorted(dirs):` here now.

Reply:
Its mostly has to do with using the list of image files that I added. Since that list is not necessarily ordered alphabetically, but instead is ordered by when I added them.

E.g.

if my list is: 
b.png
c.png

Then b will be image #0 and c will be image #1.

The if I append to my list (like when I get new data) and get
 
b.png
c.png
a.png

I don't want this sorted, otherwise b will become index #1 and c will become index #2.

Reply:
Yes, makes sense. The original `sort` at the end was only because `os.walk` is not sorted by default and I also wanted deterministic ordering, with the two sorts I proposed in the other commments we get deterministic order of filesand iterate through folders in a well defined order. If you have one list it doesn't matter, if you have two lists they will now be appended in the order as found after sorting their paths. Best of both worlds? :)